src/duration.rs

```diff
@@ -21,8 +21,8 @@ use core::fmt;
 use core::hash::{Hash, Hasher};
 use core::ops::{Add, AddAssign, Div, Mul, Neg, Sub, SubAssign};
 
-#[cfg(feature = "std")]
-use serde::{de, Deserialize, Deserializer};
+#[cfg(feature = "serde")]
```

Comment:
```suggestion
#[cfg(feature = "std")]
```

There isn't any "serde" feature in the `Cargo.toml` so I'm guessing that this line causes the serde imports to never be executed.

From Cargo.toml:

```
[features]
default = ["std"]
std = ["serde", "serde_derive"]
asn1der = ["der"]
python = ["std", "asn1der", "pyo3"]
```

When the "std" feature is enabled, it should be importing serde, but I think that you've nailed the source of the issue with your change in Duration (I'll write another comment).

Reply:
Well I'll correct myself. "Serde::Serialize" and "Deseriliaze" and also auto derivations are imported with "std"
so your current Cargo.tml content.
It would actually work if you removed these imports from std too, with I explained just above this comment

Reply:
Same thing here.. 

`rust_docs` clearly lacks some capabilities regarding crate features. They are totally undocumented in the API. Like there is absolutely no mean to know what is available besides reading the Cargo.toml..

There is a rust crate that allows documenting a crate feature, but its documentation must lie in Cargo.toml, so it will grow a little too much. And it's also one more dependency required for your crate. I think they should come with a solution to this problem in a near future